extensions/positron-code-cells/src/codeLenses.ts

```diff
@@ -33,12 +32,13 @@ export function runNextCodeLens(range: vscode.Range): vscode.CodeLens {
 
 export class CellCodeLensProvider implements vscode.CodeLensProvider {
 	provideCodeLenses(document: vscode.TextDocument): vscode.ProviderResult<vscode.CodeLens[]> {
-		if (IGNORED_SCHEMES.includes(document.uri.scheme)) {
+		const editor = getEditorFromDocument(document);
```

Comment:
I think there was a misunderstanding from my last comment. We're still doing a lossy round-trip here, starting with a `document`, trying to find its `editor` (which is not always possible since one document can be opened in many editors), and then going back to the `document`.

My proposal was to split the current `DocumentManager` class into two separate classes: one of which manages parsing of `documents` – and doesn't know about `editors`. The `CellCodeLensProvider` should only use that class. Concretely, this class would have the following methods of the current `DocumentManager`:

* `constructor`: would take a `document` – not an `editor`
* `dispose`
* `parseCells`
* `getCells`
* `getDocument`